lib/trace/otel.go

```diff
@@ -0,0 +1,250 @@
+		opts = append(opts, otlptracehttp.WithInsecure())
+	}
+	return otlptracehttp.NewClient(opts...)
+}
+
+func newGRPCClient(endpoint string, insecure bool, headers map[string]string) otlptrace.Client {
+	opts := []otlptracegrpc.Option{
+		otlptracegrpc.WithEndpoint(endpoint),
+		otlptracegrpc.WithHeaders(headers),
+	}
+	if insecure {
+		opts = append(opts, otlptracegrpc.WithInsecure())
+	}
+	return otlptracegrpc.NewClient(opts...)
+}
+
+// NewNoopTracerProvider creates a new noop TracerProvider.
+func NewNoopTracerProvider() *TracerProvider {
+	prov := trace.NewNoopTracerProvider()
+	noopShutdown := func(context.Context) error { return nil }
+
+	otel.SetTracerProvider(prov)
+
+	return &TracerProvider{
+		TracerProvider: prov,
+		shutdown:       noopShutdown,
+	}
+}
+
+// Shutdown shuts down the TracerProvider releasing any held computational resources.
+// After Shutdown is called, all methods are no-ops.
+func (tp *TracerProvider) Shutdown(ctx context.Context) error {
+	return tp.shutdown(ctx)
+}
+
+// TracerProviderFromConfigLine initializes a new TracerProvider based on the configuration
+// specified through input line.
+//
+// Supported format is: otel[=<endpoint>:<port>,<other opts>]
+// Where endpoint and port default to: 127.0.0.1:4317
+// And other opts accept:
+//   - proto: http or grpc (default).
+//   - header.<header_name>
+//
+// Example: otel=127.0.0.1:4318/v1/traces,proto=http,header.Authorization=token ***
+func TracerProviderFromConfigLine(ctx context.Context, line string) (*TracerProvider, error) {
+	params, err := tracerProviderParamsFromConfigLine(line)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewTracerProvider(ctx, params)
+}
+
+func tracerProviderParamsFromConfigLine(line string) (tracerProviderParams, error) {
+	params := defaultTracerProviderParams()
+
+	if line == "otel" {
+		// Use default params
+		return params, nil
+	}
+
+	parts := strings.SplitN(line, "=", 2)
+	if parts[0] != "otel" {
+		return params, fmt.Errorf("%w %q", ErrInvalidTracesOutput, parts[0])
+	}
+
+	tokens, err := strvals.Parse(line)
+	if err != nil {
+		return params, fmt.Errorf("error while parsing otel configuration %w", err)
+	}
+
+	for _, token := range tokens {
+		key := token.Key
+		value := token.Value
+
+		var err error
+		switch key {
+		case "otel":
+			err = params.parseURL(value)
+			if err != nil {
+				return params, fmt.Errorf("couldn't parse the otel URL: %w", err)
+			}
+		case "proto":
+			err = params.parseProto(value)
+			if err != nil {
+				return params, fmt.Errorf("couldn't parse the otel proto: %w", err)
+			}
+		default:
+			if strings.HasPrefix(key, "header.") {
+				params.parseHeader(key, value)
+				continue
+			}
+			return params, fmt.Errorf("unknown otel config key %s", key)
+		}
+	}
+
+	if params.proto == "grpc" && params.urlPath != "" {
+		return params, ErrInvalidGRPCWithURLPath
+	}
+
+	return params, nil
+}
+
+func (p *tracerProviderParams) parseURL(url string) error {
+	u, err := stdurl.Parse(url)
```

Comment:
nit: I would prefer to keep the package name isntead of the variable name

Reply:
Fixed in [88d4f84](https://github.com/grafana/k6/pull/3445/commits/88d4f84329ce50d96a0f576b826156b21a480e72).